pkg/svcat/service-catalog/broker_test.go

```diff
@@ -0,0 +1,135 @@
+			brokers, err := sdk.RetrieveBrokers()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(brokers).Should(ConsistOf(*sb, *sb2))
+			Expect(svcCatClient.Actions()[0].Matches("list", "clusterservicebrokers")).To(BeTrue())
+		})
+		It("Bubbles up errors", func() {
+			badClient := &fake.Clientset{}
+			errorMessage := "error retrieving list"
+			badClient.AddReactor("list", "clusterservicebrokers", func(action testing.Action) (bool, runtime.Object, error) {
+				return true, nil, fmt.Errorf(errorMessage)
+			})
+			sdk.ServiceCatalogClient = badClient
+			_, err := sdk.RetrieveBrokers()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring(errorMessage))
+			Expect(badClient.Actions()[0].Matches("list", "clusterservicebrokers")).To(BeTrue())
+		})
+	})
+	Describe("RetrieveBroker", func() {
+		It("Calls the generated v1beta1 List method with the passed in broker", func() {
+			broker, err := sdk.RetrieveBroker(sb.Name)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(broker.Name).To(Equal(sb.Name))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(sb.Name))
+		})
+		It("Bubbles up errors", func() {
+			brokerName := "banana"
+
+			broker, err := sdk.RetrieveBroker(brokerName)
+
+			Expect(broker).To(BeNil())
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("not found"))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(brokerName))
+		})
+	})
+	Describe("RetrieveBrokerByClass", func() {
+		It("Calls the generated v1beta1 List method with the passed in class's parent broker", func() {
+			sc := &v1beta1.ClusterServiceClass{Spec: v1beta1.ClusterServiceClassSpec{ClusterServiceBrokerName: sb.Name}}
+			broker, err := sdk.RetrieveBrokerByClass(sc)
+
+			Expect(broker).NotTo(BeNil())
+			Expect(err).NotTo(HaveOccurred())
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(sb.Name))
+		})
+
+		It("Bubbles up errors", func() {
+			brokerName := "banana"
+			sc := &v1beta1.ClusterServiceClass{Spec: v1beta1.ClusterServiceClassSpec{ClusterServiceBrokerName: brokerName}}
+			broker, err := sdk.RetrieveBrokerByClass(sc)
+
+			Expect(broker).To(BeNil())
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("not found"))
+			actions := svcCatClient.Actions()
+			Expect(actions[0].Matches("get", "clusterservicebrokers")).To(BeTrue())
+			Expect(actions[0].(testing.GetActionImpl).Name).To(Equal(brokerName))
+		})
+	})
+	Describe("Sync", func() {
+		It("Useds the generated b1beta1 Retrieve method to get the broker, and then updates it with a new RelistRequests", func() {
```

Comment:
This testcase really highlights where the output based tests aren't enough to validate the svcat client. 👍